src/common/hooks/useDocuNinjaActions.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { useQueryClient } from 'react-query';
+import { useSetAtom } from 'jotai';
+import { request } from '$app/common/helpers/request';
+import { endpoint } from '$app/common/helpers';
+import { resetDocuNinjaDataAtom } from '$app/common/atoms/docuninja';
+
+// This hook ONLY provides actions, no data fetching
+export function useDocuNinjaActions() {
+  const queryClient = useQueryClient();
+  const resetData = useSetAtom(resetDocuNinjaDataAtom);
+
+  const getToken = (): string => {
+    return localStorage.getItem('X-DOCU-NINJA-TOKEN') || '';
+  };
+
+  const createAccount = async () => {
+    try {
+      await request(
+        'POST',
+        endpoint('/api/docuninja/create'),
+        {},
+        { skipIntercept: true }
+      );
+      
+      // Refetch the data after creating account
+      await queryClient.invalidateQueries(['/api/docuninja/login']);
+    } catch (error) {
+      throw error;
+    }
+  };
```

Comment:
If you use promise-based style with .then .catch error handling becomes much easier as you can handle exceptions with types.